src/Swift.Runtime/src/Metadata/TypeMetadata.cs

```diff
@@ -249,4 +253,113 @@ public override int GetHashCode ()
     /// Gets the type metadata cache for the runtime.
     /// </summary>
     public static ITypeMetadataCache Cache => cache;
-}
\ No newline at end of file
+
+    /// <summary>
+    /// Attempt to get the Swift type metadata for the given object instance
+    /// </summary>
+    /// <typeparam name="T">The type of the object</typeparam>
+    /// <param name="result">The result of looked up type metadata</param>
+    /// <returns>true on success false otherwise</returns>
+    public static bool TryGetTypeMetadata<T>([NotNullWhen(true)] out TypeMetadata? result)
+    {
+        if (cache.TryGet(typeof(T), out result))
+            return true;
+        return TryGetTypeMetadataUncached<T>(out result);
+    }
+
+    /// <summary>
+    /// Attempt to get the Swift type metadata for the given type
+    /// </summary>
+    /// <typeparam name="T">The type of the object</typeparam>
+    /// <returns>The result of the looked up type metadata on success</returns>
+    /// <exception cref="Exception">Throws when lookup fails</exception>
+    public static TypeMetadata GetTypeMetadataOrThrow<T>()
```

Comment:
I would drop the `OrThrow`. The default semantic for functions in .NET is that they can throw. The `Try` semantics above indicate the non-throwing nature.